src/Libraries/RevitNodes/Filter/FilterRule.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using Autodesk.Revit.DB;
+using DynamoServices;
+using Autodesk.DesignScript.Runtime;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using System.Collections.Generic;
+
+namespace Revit.Filter
+{
+    /// <summary>
+    /// Revit Filter Rule
+    /// </summary>
+    public class FilterRule
+    {
+
+        #region Internal Properties
+
+        /// <summary>
+        /// Internal reference to the Revit Element
+        /// </summary>
+        internal Autodesk.Revit.DB.FilterRule InternalFilterRule
+        { 
+            get; set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        /// <param name="internalFilterRule"></param>
+        internal FilterRule(Autodesk.Revit.DB.FilterRule internalFilterRule)
+        {
+            this.InternalFilterRule = internalFilterRule;
+        }
+
+        #endregion
+
+        #region Helpers
+
+        /// <summary>
+        /// FilterType Enumeration
+        /// </summary>
+        [IsVisibleInDynamoLibrary(false)]
+        public enum FilterType
+        {
+            BeginsWith,
+            Contains,
+            EndsWith,
+            Equals,
+            Greater,
+            Less,
+            GeaterOrEqual,
+            LessOrEqual,
+            NotBeginsWith,
+            NotContains,
+            NotEndsWith,
+            NotEquals,
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Create a new Filter Rule
+        /// </summary>
+        /// <param name="type">Filter Rule Type</param>
+        /// <param name="value">Value to check</param>
+        /// <param name="parameter">Parameter to filter</param>
+        /// <returns></returns>
+        public static FilterRule ByRuleType(string typename, object value, Elements.Parameter parameter)
+        {
+
+            FilterType type = FilterType.Contains;
+
+            if (!Enum.TryParse<FilterType>(typename, out type))
+            {
+                throw new Exception(Properties.Resources.InvalidFilterType);
+            }
+
+            ElementId parameterId = parameter.InternalParameter.Id;
+
+            // assemble the method name to construct a new filter using the FilterType
+            string methodname = "Create" + type.ToString() + "Rule";
+    
+            // all of the following FilterType construcotrs are handled the same
+            if (
+                type == FilterType.Equals ||
+                type == FilterType.NotEquals ||
+                type == FilterType.Greater ||
+                type == FilterType.Less ||
+                type == FilterType.GeaterOrEqual ||
+                type == FilterType.LessOrEqual
+                )
+            {
+                if (value.GetType() == typeof(int))
```

Comment:
The other solution would be to write every call explicitly, but it would be a switch over at least 18 cases.